.github/workflows/checks.yml

```diff
@@ -0,0 +1,26 @@
+name: Checks
+
+on:
+  pull_request:
+    branches: [main]
+
+jobs:
+  test_and_lint:
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Cache/Restore Mint packages
+        id: mint-cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ github.workspace }}/mint
+          key: ${{ runner.os }}-mint-${{ hashFiles('**/Mintfile') }}
+          restore-keys: ${{ runner.os }}-mint-
+      - name: Install package manager "Mint"
+        run: |
+          brew upgrade mint || brew install mint || true
```

Comment:
Not sure this is a good idea. I'd suggest to check if mint is installed first, and then try to upgrade or install (even better - force specific, fixed version in the process)